apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -206,7 +213,7 @@ defmodule OMG.Watcher.BlockGetter do
 
       _ = Logger.debug("Synced height is #{inspect(synced_height)}, got #{length(blocks_to_apply)} blocks to apply")
 
-      Enum.each(blocks_to_apply, &GenServer.cast(__MODULE__, {:apply_block, &1}))
+      Enum.each(blocks_to_apply, &apply_block(&1))
```

Comment:
reason for change?

Reply:
GenServer.cast/GenServer.call with decorate are very tricky :D 
I will restore the use of GenServer.cast

Reply:
why not call a local function?

Reply:
I had the same dilemma :D
1. GenServer timer, it can fire if whole apply_block will make here
2. GenServer approach is to make small task
3. supposedly, the message queue is quite fast (erlangu / elixir puts quite a lot of effort in it )

Reply:
not sure I understand

Reply:
let me jump in, as I just answered the same question :).

The reason for this is that we want to handle block application on a per-block basis. Mainly it is for more granularity - application of every block takes the getter from one valid state to another, so we wanted to replicate that in the GenServer messages.

Also a long range of large blocks to apply would result in the call to `sync` block the Process for a long time, chunking it into blocks keeps it more responsive - eg. after we received and sent some blocks to `apply_block`, we might want to immediately start pulling new ones (I think this is how this works, but not 100% sure)

There's also some historical reasons - we used to `stop` this `GenServer` under some logic-driven circumstances, this idea was rightfully abandoned, but the splitting of block application remains.

Reply:
Not arguing with the logical implementation: just how the measurement was done :)